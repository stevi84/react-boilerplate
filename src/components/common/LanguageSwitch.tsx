import { Box, FormControl, InputLabel, MenuItem, Select, SelectChangeEvent } from '@mui/material';
import { useTranslation } from 'react-i18next';

export const LanguageSwitch = () => {
  const { t, i18n } = useTranslation();

  const onChange = (event: SelectChangeEvent) => i18n.changeLanguage(event.target.value);

  return (
    <Box sx={{ minWidth: 120 }}>
      <FormControl fullWidth>
        <InputLabel>{t('language')}</InputLabel>
        <Select value={i18n.language} label={t('language')} onChange={onChange}>
          <MenuItem value={'de'}>{t('de')}</MenuItem>
          <MenuItem value={'en'}>{t('en')}</MenuItem>
        </Select>
      </FormControl>
    </Box>
  );
};
