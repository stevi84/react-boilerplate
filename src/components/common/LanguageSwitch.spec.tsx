import { describe, it, expect, vi } from 'vitest';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import { fireEvent, render, screen, within } from '@testing-library/react';
import { LanguageSwitch } from './LanguageSwitch';

const { changeLanguageMock } = vi.hoisted(() => ({
  changeLanguageMock: vi.fn(),
}));
vi.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str: string) => str,
    i18n: {
      language: 'de',
      changeLanguage: changeLanguageMock,
    },
  }),
}));

describe('LanguageSwitch', () => {
  it('should equal saved snapshot', () => {
    const renderer: ShallowRenderer = createRenderer();
    renderer.render(<LanguageSwitch />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('should change the language', () => {
    render(<LanguageSwitch />);
    expect(changeLanguageMock.mock.calls.length).toEqual(0);
    fireEvent.mouseDown(screen.getByText('de'));
    const listbox = within(screen.getByRole('listbox'));
    fireEvent.click(listbox.getByText('en'));
    expect(changeLanguageMock.mock.calls.length).toEqual(1);
    expect(changeLanguageMock.mock.calls[0][0]).toEqual('en');
  });
});
