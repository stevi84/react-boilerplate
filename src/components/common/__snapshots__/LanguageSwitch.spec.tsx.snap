// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LanguageSwitch > should equal saved snapshot 1`] = `
<ForwardRef(Box)
  sx={
    {
      "minWidth": 120,
    }
  }
>
  <ForwardRef(FormControl)
    fullWidth={true}
  >
    <ForwardRef(InputLabel)>
      language
    </ForwardRef(InputLabel)>
    <ForwardRef(Select)
      label="language"
      onChange={[Function]}
      value="de"
    >
      <ForwardRef(MenuItem)
        value="de"
      >
        de
      </ForwardRef(MenuItem)>
      <ForwardRef(MenuItem)
        value="en"
      >
        en
      </ForwardRef(MenuItem)>
    </ForwardRef(Select)>
  </ForwardRef(FormControl)>
</ForwardRef(Box)>
`;
